boring/src/pkey.rs

```diff
@@ -260,6 +280,26 @@ where
         private_key_to_der_pkcs8_passphrase,
         ffi::i2d_PKCS8PrivateKey_bio
     }
+
+    /// Outputs a copy of the "raw" form of the private key. Only supported for certain key types.
+    ///
+    /// Returns the number of bytes written, or the size of the raw form if `out` is empty.
+    #[corresponds(EVP_PKEY_get_raw_private_key)]
+    pub fn raw_private_key(&self, out: &mut [u8]) -> Result<usize, ErrorStack> {
```

Comment:
I would rather have two methods, `raw_private_key` and `raw_private_key_len`, rather than piggyback `raw_private_key` to do a different thing when `out` is empty.